src/mgds/pipelineModules/DropTags.py

```diff
@@ -0,0 +1,155 @@
+            delimiter_in_name: str,
+            regex_enabled_in_name: str,     #whether to process special tags with regex
+            keep_tags_count_in_name: str,
+            text_out_name: str,
+    ):
+        super(DropTags, self).__init__()
+        self.text_in_name = text_in_name
+        self.enabled_in_name = enabled_in_name
+        self.probability_in_name = probability_in_name
+        self.dropout_mode_in_name = dropout_mode_in_name
+        self.special_tags_in_name = special_tags_in_name
+        self.special_tag_mode_in_name = special_tag_mode_in_name
+        self.delimiter_in_name = delimiter_in_name
+        self.regex_enabled_in_name = regex_enabled_in_name
+        self.keep_tags_count_in_name = keep_tags_count_in_name
+        self.text_out_name = text_out_name
+
+    def length(self) -> int:
+        return self._get_previous_length(self.text_in_name)
+
+    def get_inputs(self) -> list[str]:
+        return [self.text_in_name, self.enabled_in_name, self.probability_in_name, self.dropout_mode_in_name, self.special_tags_in_name, self.special_tag_mode_in_name, self.delimiter_in_name, self.keep_tags_count_in_name]
+
+    def get_outputs(self) -> list[str]:
+        return [self.text_out_name]
+    
+    #convert special_tags to list depending on whether it's a newline-separated csv/txt file or a delimiter-separated string
+    #cached to reduce file read operations
+    @functools.lru_cache
+    def get_special_tags(self, sptags, delim):
+        if (sptags.endswith(".txt") and os.path.isfile(sptags)):
+            with open(sptags) as special_tags_file:
+                return [line.rstrip('\n') for line in special_tags_file]
+        elif (sptags.endswith(".csv") and os.path.isfile(sptags)):
+            with open(sptags, 'r') as special_tags_file:
+                splist = []
+                for row in csv.reader(special_tags_file):
+                    splist.append(row[0])
+                return splist
+        else:
+            return [tag.strip() for tag in sptags.split(delim)]
+
+    #parse regex expressions, create new special list based on matches   
+    def parse_regex(self, splist_in, taglist):
+        splist_out = []
+        regex_spchars = set(".^$*+?!\{\}\[\]|()\\")
+        for c in splist_in:
+            if any((a in regex_spchars) for a in c):    #only do regex matching if tag contains special character
+                c = c.replace("\)", "\\\\\)")
+                c = c.replace("\(", "\\\\\(")   #hopefully fix issues caused by "\(\)" syntax without affecting other regex
+                r = re.compile(c)
+                for s in taglist:
+                    if r.fullmatch(s):
+                        splist_out.append(s)
+            else:
+                splist_out.append(c)
+        return splist_out    
+    
+    #change probability evaluated against random() depending on mode
+    def probability_weighted(self, p, mode, i, len):
+        if mode == "RANDOM":
+            return float(p)
```

Comment:
p should already be a float. And if it isn't, some of the other tests wouldn't work

Reply:
I guess not strictly necessary, but included that to reduce the possibility of a problem caused by an "improper" input. Will see if I can find a cleaner way to do that